### PR TITLE
Feature/autosuggest section header

### PR DESCRIPTION
**Pull request by manthachan**
** Please provide as much detail as possible, but feel free to remove irrelevant sections **

{{ Reason for change }}
Added the section header for the autosuggest component

### CODE REVIEW

File: react/Autosuggest/Autosuggest.less [hunk 1]
```diff
@@ -62,23 +62,32 @@
...
   }
 }
 
+.sectionTitle_fullWidth {
+  @media @mobile {
+    position: relative;
+    left: -50vw;
+    background: @saWhite;
+    margin: 0;
+    padding: 0 (@grid-gutter-width * 0.75);
```
Comment (mengtzu) [inline react/Autosuggest/Autosuggest.less]: As discussed please change this to either some appropriate multiplier of `@grid-base` or just hardcode it.

Also concur with ZY that this should be mobile first

File: react/Autosuggest/Autosuggest.less [hunk 2]
```diff
@@ -62,28 +62,37 @@
...
+  background: @saWhite;
   .sectionContainer:first-child & {
     margin-top: 0;
   }
 }
 
+.sectionTitle_fullWidth {
+  @media @mobile {
+    position: relative;
+    left: -50vw;
```
Comment (kosanna) [inline react/Autosuggest/Autosuggest.less]: Can't we reuse suggestions list styles here?

**Comment by kosanna** [comment]
@ZhenYongSeekAsia How the section header looks like? ;)

**Comment by kosanna** [comment]
Is that by design that Section labels are not following the standard padding?
Screen Shot 2019-05-20 at 5 43 33 pm

**Comment by patrickliu-sa** [comment]
Section header should share the same padding as regular list items.
Is it something equivalent to `<optgroup>` in `HTML` which shouldn't be selectable?

**Comment by manthachan** [comment]
> Section header should share the same padding as regular list items.
> Is it something equivalent to `<optgroup>` in `HTML` which shouldn't be selectable?

actually the section header can be a component, for the current usage there will be a clear button inside the section header, so the selectable should be control in the application but not style guide

**Pull request merged**